component/azstorage/datalake.go

```diff
@@ -461,17 +462,31 @@ func (dl *Datalake) List(prefix string, marker *string, count int32) ([]*interna
 
 	// Process the paths returned in this result segment (if the segment is empty, the loop body won't execute)
 	for _, pathInfo := range listPath.Paths {
-		mode, err := getFileMode(*pathInfo.Permissions)
-		if err != nil {
-			log.Err("Datalake::List : Failed to get file mode for %s [%s]", *pathInfo.Name, err.Error())
-			m := ""
-			return pathList, &m, err
+		var mode fs.FileMode
+		if pathInfo.Permissions != nil {
+			mode, err = getFileMode(*pathInfo.Permissions)
+			if err != nil {
+				log.Err("Datalake::List : Failed to get file mode for %s [%s]", *pathInfo.Name, err.Error())
+				m := ""
+				return pathList, &m, err
+			}
+		} else {
+			// This happens when a blob account is mounted with type:adls
```

Comment:
why are we adding this condition?

Reply:
In case of wrong configuration blobfuse2 binary is crashing (coring out). In any case we shall just go down gracefully or generate error logs. Crash is not a good thing so trying to avoid it here.